panel/param.py

```diff
@@ -357,8 +358,10 @@ def _get_widgets(self):
         filtered = [(k,p) for (k,p) in sorted_precedence
                     if ((p.precedence is None) or (p.precedence >= self.display_threshold))]
         groups = itertools.groupby(filtered, key=key_fn)
-        sorted_groups = [sorted(grp) for (k,grp) in groups]
-        ordered_params = [el[0] for group in sorted_groups for el in group]
+        # Params preserve definition order in Python 3.6+
+        ordered_params = sys.version_info.major == 3 and sys.version_info.minor >= 6
+        ordered_groups = [list(grp) if ordered_params else sorted(grp) for (k,grp) in groups]
+        ordered_params = [el[0] for group in ordered_groups for el in group]
```

Comment:
One more change: it's confusing to use ordered_params as both the boolean flag and as the set of ordered params.  Maybe rename the first one to `params_already_ordered`?

Reply:
I'd go with ``if dict_ordered``.

Reply:
Fixed in 791778d